### PR TITLE
Search screen

### PR DESCRIPTION
**Pull request by ndt0409**
## Related Tickets
- [#Task 53278: Create search movie by genres,categories, top rated list](https://edu-redmine.sun-asterisk.vn/issues/53278)
## WHAT
- Create UI search
- Search movie by genres, categories, top rated list
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/ndt/themoviedb/ui/details/MovieDetailsFragment.kt [hunk 1]
```diff
@@ -152,6 +157,21 @@ class MovieDetailsFragment :
         }
     }
 
+    private fun initToolBar() {
+        toolbar_base?.let {
```
Comment (minhnn-1279) [inline app/src/main/java/com/ndt/themoviedb/ui/details/MovieDetailsFragment.kt]: cái này a dùng binding nhé a

File: app/src/main/java/com/ndt/themoviedb/ui/details/adapter/CastAdapter.kt [hunk 2]
```diff
@@ -16,8 +16,7 @@ class CastAdapter(var onItemClick: (Cast, Int) -> Unit = { _, _ -> }) :
 
     private var binding: ItemCastBinding? = null
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder
-    {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
         binding =
             ItemCastBinding.inflate(LayoutInflater.from(parent.context), parent, false)
         return ViewHolder(binding!!, onItemClick)
```
Comment (minhnn-1279) [inline app/src/main/java/com/ndt/themoviedb/ui/details/adapter/CastAdapter.kt]: tránh assert not null nhé a

File: app/src/main/java/com/ndt/themoviedb/ui/search/SearchFragment.kt [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            viewBinding.recyclerTopRate.adapter = topRatedAdapter.apply {
+                onItemClick = { movie, _ ->
+                    addFragment(MovieDetailsFragment.getInstance(movie.id, movie.title))
+                }
+            }
+        }
+    }
+
+    private fun initRefresh() {
+        view?.search_swipe_refresh?.setOnRefreshListener {
```
Comment (minhnn-1279) [inline app/src/main/java/com/ndt/themoviedb/ui/search/SearchFragment.kt]: chỗ này cx dùng binding nhé a

File: app/src/main/java/com/ndt/themoviedb/ui/search/adapter/CategoryAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.ndt.themoviedb.ui.search.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.ndt.themoviedb.data.model.Category
+import com.ndt.themoviedb.databinding.ItemCategoryBinding
+import com.ndt.themoviedb.ui.base.BaseAdapter
+import com.ndt.themoviedb.ui.base.BaseViewHolder
+
+class CategoryAdapter(var onItemClick: (Category, Int) -> Unit = { _, _ -> }) :
```
Comment (minhnn-1279) [inline app/src/main/java/com/ndt/themoviedb/ui/search/adapter/CategoryAdapter.kt:10]: cái onclick này a để val nhé

File: app/src/main/java/com/ndt/themoviedb/ui/details/MovieDetailsFragment.kt [hunk 5]
```diff
@@ -19,9 +19,11 @@ import com.ndt.themoviedb.ui.base.BaseFragment
 import com.ndt.themoviedb.ui.details.adapter.CastAdapter
 import com.ndt.themoviedb.ui.details.adapter.ProduceAdapter
 import com.ndt.themoviedb.ui.details.adapter.TrailerAdapter
+import com.ndt.themoviedb.ui.mainscreen.MainActivity
 import com.ndt.themoviedb.ui.utils.NetworkUtil
 import com.ndt.themoviedb.ui.utils.constant.UrlConstant
 import com.ndt.themoviedb.ui.utils.extension.addFragment
+import kotlinx.android.synthetic.main.toolbar_base.*
```
Comment (daolq-2712) [inline app/src/main/java/com/ndt/themoviedb/ui/details/MovieDetailsFragment.kt]: remove dòng này em ơi,
mình dùng viewBinding

File: app/src/main/java/com/ndt/themoviedb/ui/details/MovieDetailsFragment.kt [hunk 6]
```diff
@@ -152,6 +155,21 @@ class MovieDetailsFragment :
         }
     }
 
+    private fun initToolBar() {
```
Comment (daolq-2712) [inline app/src/main/java/com/ndt/themoviedb/ui/details/MovieDetailsFragment.kt]: move phương thức này lên dòng 70 e nhé!

ref: https://github.com/framgia/coding-standards/blob/master/vn/android/coding_convention_android_kotlin.md#class-layout

File: app/src/main/java/com/ndt/themoviedb/ui/details/adapter/CastAdapter.kt [hunk 7]
```diff
@@ -35,17 +30,16 @@ class CastAdapter(var onItemClick: (Cast, Int) -> Unit = { _, _ -> }) :
         }
 
         private fun getImageCircle(cast: Cast?) {
-            GetImageAsyncTask(
-                object : OnFetchImageListener {
+            GetImageAsyncTask(object : OnFetchImageListener {
```
Comment (daolq-2712) [inline app/src/main/java/com/ndt/themoviedb/ui/details/adapter/CastAdapter.kt:33]: như này sẽ gây leak memory

@vietnh-1109 hỗ trợ bạn hộ anh!

File: app/src/main/java/com/ndt/themoviedb/ui/listmovie/ListMovieFragment.kt [hunk 8]
```diff
@@ -130,6 +136,21 @@ class ListMovieFragment : BaseFragment<FragmentListMovieBinding>(FragmentListMov
         }
     }
 
+    private fun initToolBar() {
```
Comment (daolq-2712) [inline app/src/main/java/com/ndt/themoviedb/ui/listmovie/ListMovieFragment.kt]: move lên dòng 64

**Review by daolq-2712 — CHANGES_REQUESTED** [review]
Pls check my comments!

**Pull request merged**